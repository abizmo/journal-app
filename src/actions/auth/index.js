export { default as login } from './login';
export { default as loginWithEmail } from './loginWithEmail';
export { default as loginWithGoogle } from './loginWithGoogle';
export { default as logout } from './logout';
export { default as signOut } from './signOut';
export { default as signUp } from './signUp';

export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

export const types = {
  LOGIN,
  LOGOUT,
};
