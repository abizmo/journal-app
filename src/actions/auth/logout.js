import { LOGOUT } from ".";

const logout = () => ({
  type: LOGOUT,
});

export default logout;
