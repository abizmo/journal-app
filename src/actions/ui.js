export const CLEAR_ERROR = 'CLEAR_ERROR';
export const SET_ERROR = 'SET_ERROR';
export const START_ACTION = 'START_ACTION';
export const STOP_ACTION = 'STOP_ACTION';

export const types = {
  CLEAR_ERROR,
  SET_ERROR,
  START_ACTION,
  STOP_ACTION,
};

export const clearError = () => ({
  type: CLEAR_ERROR,
});

export const setError = (msgError) => ({
  type: SET_ERROR,
  payload: msgError,
});

export const startAction = () => ({
  type: START_ACTION,
});

export const stopAction = () => ({
  type: STOP_ACTION,
});
