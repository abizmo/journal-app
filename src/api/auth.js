import {
  createUserWithEmailAndPassword,
  onAuthStateChanged,
  signInWithEmailAndPassword,
  signInWithPopup,
  signOut as firebaseSignOut,
  updateProfile,
} from "firebase/auth";

import { auth, provider } from "../firebase";

export const onCurrentUserChanged = (cb) => onAuthStateChanged(auth, cb);
  
export const signIn = async ({ email, password }) => {
  const { user } = await signInWithEmailAndPassword(auth, email, password)

  return {
    name: user.displayName,
    userId: user.uid
  };
};

export const signInWithGoogle = () => signInWithPopup(auth, provider)
  .then(({ user }) => ({
    name: user.displayName,
    userId: user.uid
}));

export const signOut = () => firebaseSignOut(auth);

export const signUpWithEmailAndPassword = ({
  email,
  name,
  password
}) => createUserWithEmailAndPassword(auth, email, password)
  .then(({ user }) => updateProfile(user, { displayName: name })
  .then(() => ({ name, userId: user.uid})));
