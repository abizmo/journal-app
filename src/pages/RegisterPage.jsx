import React from 'react';
import { Link } from 'react-router-dom';

const RegisterPage = () => (
  <>
    <h3 className="auth__title mb-4">Register</h3>
    <form>
      <input
        autoComplete="none"
        className="auth__input mb-2"
        name="name"
        placeholder="Your Name"
        required
        type="text"
      />
      <input
        autoComplete="off"
        className="auth__input mb-2"
        name="email"
        placeholder="tu@email.com"
        required
        type="email"
      />
      <input
        className="auth__input mb-2"
        name="password"
        placeholder="TuContraseÃ±@"
        required
        type="password"
      />
      <input
        className="auth__input mb-2"
        name="confirm-password"
        placeholder="TuContraseÃ±@"
        required
        type="password"
      />
      <button
        className="btn btn-block btn-primary mb-2"
        type="submit"
      >Register</button>
      <Link
        className="link d-block"
        to="/auth/login"
      >Already register?</Link>
    </form>
  </>
);

export default RegisterPage;
