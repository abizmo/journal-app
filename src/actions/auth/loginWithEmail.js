import { login } from ".";

import { startAction, stopAction } from "../ui";
import { signIn } from "../../api/auth";
import { error } from "../../helpers/alert";

const loginWithEmail = ({ email, password }) => async (dispatch) => {
  try {
    dispatch(startAction());
    const user = await signIn({ email, password });
    dispatch(login(user));
  } catch ({ message }) {
    error(message);
  }
  dispatch(stopAction());
};

export default loginWithEmail;
