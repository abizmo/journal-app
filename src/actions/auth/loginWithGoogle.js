import { login } from ".";

import { signInWithGoogle } from "../../api/auth";
import { error } from "../../helpers/alert";

const loginWithGoogle = () => async (dispatch) => {
  try {
    const user = await signInWithGoogle();
    dispatch(login(user));
  } catch ({ message }) {
    error(message);
  }
};

export default loginWithGoogle;
