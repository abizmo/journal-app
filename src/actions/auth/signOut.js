import { logout } from '.';

import { clearNotes } from '../notes';
import { signOut as signOutApi } from '../../api/auth';
import { error } from "../../helpers/alert";

const signOut = () => (dispatch) => {
  signOutApi()
    .then(() => {
      dispatch(logout());
      dispatch(clearNotes());
    })
    .catch(({ message }) => error(message));
};

export default signOut;
