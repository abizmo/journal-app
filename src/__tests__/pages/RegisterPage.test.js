import { mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import createMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { signUp } from "../../actions/auth";
import { setError } from "../../actions/ui";
import RegisterPage from "../../pages/RegisterPage"
jest.mock('../../actions/auth');
jest.mock('../../actions/ui');

const initialState = {
  ui: {
    error: null,
  },
};
const middlewares = [thunk];
const mockStore = createMockStore(middlewares);

describe('<RegisterPage />', () => {
  const store = mockStore(initialState);
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={ store }>
      <MemoryRouter>
        <RegisterPage />
      </MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    store.clearActions();
    jest.resetAllMocks();
  })
  test('should render and match snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

  test('should dispatch setError', () => {
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    expect(setError).toHaveBeenCalled();
  });

  test('should dispatch signUp', () => {
    const name = 'goccita';
    const email = 'goccita@abizmo.dev';
    const password = '12345678';
    wrapper.find('#name').simulate('change', { target: { name: 'name', value: name } });
    wrapper.find('#email').simulate('change', { target: { name: 'email', value: email } });
    wrapper.find('#password').simulate('change', { target: { name: 'password', value: password } });
    wrapper.find('#confirmPassword').simulate('change', { target: { name: 'confirmPassword', value: password } });
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    
    expect(signUp).toHaveBeenCalledWith({ name, email, password });
  });
});
