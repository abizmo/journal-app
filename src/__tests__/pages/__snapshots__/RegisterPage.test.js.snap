// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage /> should render and match snapshot 1`] = `
Array [
  <h3
    className="auth__title mb-4"
  >
    Register
  </h3>,
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="none"
      className="auth__input mb-2"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Your Name"
      required={true}
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input mb-2"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="tu@email.com"
      required={true}
      type="email"
      value=""
    />
    <input
      className="auth__input mb-2"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="TuContraseñ@"
      required={true}
      type="password"
      value=""
    />
    <input
      className="auth__input mb-2"
      id="confirmPassword"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="TuContraseñ@"
      required={true}
      type="password"
      value=""
    />
    <button
      className="btn btn-block btn-primary mb-2"
      type="submit"
    >
      Register
    </button>
    <a
      className="link d-block"
      href="/auth/login"
      onClick={[Function]}
    >
      Already register?
    </a>
  </form>,
]
`;
